lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -481,22 +482,24 @@ def _alert_rules_error(self, event):
 DEFAULT_ALERT_RULES_RELATIVE_PATH = "./src/loki_alert_rules"
 DEFAULT_LOG_PROXY_RELATION_NAME = "log-proxy"
 
-PROMTAIL_BINARY_GZIP_URL = "https://github.com/canonical/loki-k8s-operator/releases/download/promtail-v2.5.0/promtail-static-amd64.gz"
-
+PROMTAIL_BASE_URL = "https://github.com/canonical/loki-k8s-operator/releases/download"
+# To update Promtail version you only need to change the PROMTAIL_VERSION and
+# update all sha256 sums in PROMTAIL_BINARIES. To support a new architecture
+# you only need to add a new key value pair for the architecture in PROMTAIL_BINARIES.
+PROMTAIL_VERSION = "v2.5.0"
+PROMTAIL_BINARIES = {
+    "amd64": {
+        "filename": "promtail-static-amd64",
+        "zipsha": "543e333b0184e14015a42c3c9e9e66d2464aaa66eca48b29e185a6a18f67ab6d",
+        "binsha": "17e2e271e65f793a9fbe81eab887b941e9d680abe82d5a0602888c50f5e0cac9",
+    },
+}
```

Comment:
I thought the point of this PR was to support arm64?

Reply:
yes but at present this is not arm64 v2.5.0 binary available in our repo. It is a simple matter to add one more key value pair to this dictionary when we upload those binaries.

Reply:
As I understood this refactor was to bring in "support" for multiple architectures. We can actually add additional architectures trivially subsequently.